apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/components/FiatAmountInput.tsx

```diff
@@ -0,0 +1,105 @@
+import {useCallback, useRef, useState} from 'react'
+import {type TextInput} from 'react-native'
+import {type PrimitiveAtom, useAtom, useAtomValue, useSetAtom} from 'jotai'
+import {
+  btcOrSatAtom,
+  feeAmountAtom,
+  SATOSHIS_IN_BTC,
+  tradeBtcPriceAtom,
+} from '../atoms'
+import {checkInputValueIsDecimal} from '../../../utils'
+import {Stack, Text} from 'tamagui'
+import AmountInput from './AmountInput'
+import {currencies} from '../../../../../utils/localization/currency'
+import {offerForTradeChecklistAtom} from '../../../atoms'
+import CalculatedWithLiveRate from './CalculatedWithLiveRate'
+
+interface Props {
+  automaticCalculationDisabled?: boolean
+  fiatValueAtom: PrimitiveAtom<string | undefined>
+  btcValueAtom: PrimitiveAtom<string | undefined>
+  showSubtitle?: boolean
+}
+
+function FiatAmountInput({
+  automaticCalculationDisabled,
+  btcValueAtom,
+  fiatValueAtom,
+  showSubtitle,
+}: Props): JSX.Element {
+  const ref = useRef<TextInput>(null)
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+
+  const [fiatValue, setFiatValue] = useAtom(fiatValueAtom)
+  const setBtcValue = useSetAtom(btcValueAtom)
+  const feeAmount = useAtomValue(feeAmountAtom)
+  const offerForTradeChecklist = useAtomValue(offerForTradeChecklistAtom)
+  const tradeBtcPrice = useAtomValue(tradeBtcPriceAtom)
+  const btcOrSat = useAtomValue(btcOrSatAtom)
+
+  const calculateBtcValueOnFiatAmountChange = useCallback(
```

Comment:
Why not make this into atom?

Reply:
Great work moving all callbacks to actionAtoms. It is much more readable when all updates are written in one way.